projects/plugins/boost/app/lib/class-speed-score.php

```diff
@@ -130,11 +155,73 @@ public function fetch_speed_score( $request ) {
 
 		// If this is a fresh install, there might not be any speed score history. In which case, we want to fetch the initial scores.
 		$history = new Speed_Score_History( $url );
-		if ( null === $history->latest() ) {
+		if ( null === $history->latest() && empty( $score_request ) ) {
 			return $this->dispatch_speed_score_request( $request );
 		}
 
-		return $this->prepare_speed_score_response( $url, $score_request );
+		return $this->prepare_speed_score_response( $url, $score_request, $score_request_no_boost );
+	}
+
+	/**
+	 * If it is time to fetch the score without boost, fetch it.
+	 *
+	 * @param string $url Url of the site.
+	 *
+	 * @return Speed_Score_Request
+	 */
+	private function maybe_dispatch_previous_score_request( $url ) {
+
+		// Allow `jb-disable-module` URL param to fetch score without boost modules being active.
+		add_filter( 'jetpack_boost_excluded_query_parameters', array( $this, 'allow_jb_disable_module' ) );
+
+		$url_no_boost = $this->get_boost_modules_disabled_url( $url );
+
+		$history       = new Speed_Score_History( $url_no_boost );
+		$latest_score  = $history->latest( 0, true );
+		$score_request = $this->get_score_request_by_url( $url_no_boost );
+
+		// Refetch the score without boost if it is older than a day.
+		if ( ( empty( $score_request ) || ! $score_request->is_pending() ) && ( null === $latest_score || $latest_score['timestamp'] < strtotime( '- 24 hours' ) ) ) {
+			$score_request = new Speed_Score_Request( $url_no_boost ); // Dispatch a new speed score request to measure score without boost.
+			$score_request->store( 1800 ); // Keep the request for 30 minutes even if no one access the results.
```

Comment:
If the [back end keeps the result for up to an hour](https://opengrok.a8c.com/source/xref/trunk/wp-content/lib/jetpack-boost/speed-score-state.php?r=232390#127), we should probably cache the request on the user's server for an hour.

It increases the chance that users who browse away from Boost while partway through generating speed scores will get the same "no-boost" speed score they already generated, instead of firing another request.

In fact, I'm tempted to suggest we bump both cache durations to more like 12 hours for that reason. But that can be done in a separate PR / step.